### PR TITLE
Closes #1539 kebab caseificiation

### LINKED ISSUE #1539: Fix kebab-caseificiation
See https://github.com/disneystreaming/smithy4s/blame/89e88c972174fa0a3b2853901a52a5a6a4662c6d/modules/decline/src/core/commons.scala#L31

Running this on a string that starts with a lowercase erases the first letter.

### PR DESCRIPTION
**Pull request by dwalend**
## PR Checklist (not all items are relevant to all PRs)

- [x] Added unit-tests (for runtime code)
- [ ] Added bootstrapped code + smoke tests (when the rendering logic is modified)
- [ ] Added build-plugins integration tests (when reflection loading is required at codegen-time)
- [ ] Added alloy compliance tests (when simpleRestJson protocol behaviour is expanded/updated)
- [ ] Updated dynamic module to match generated-code behaviour
- [ ] Added documentation
- [x] Updated changelog

### CODE REVIEW

**Comment by dwalend** [comment]
It doesn't look like github linked this PR back to https://github.com/disneystreaming/smithy4s/issues/1539 . What's the right way to make that happen?

**Comment by kubukoz** [comment]
> It doesn't look like github linked this PR back to #1539 . What's the right way to make that happen?

I think you need a hash in the PR title. Ideally, you can say "Closes #1539" and it'll link them so that when the PR gets merged, the issue gets closed.

File: modules/decline/test/src/smithy4s/decline/core/commonsSpec.scala [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package smithy4s.decline.core
+
+import weaver._
+
+object commonsSpec extends FunSuite {
```
Comment (kubukoz) [inline modules/decline/test/src/smithy4s/decline/core/commonsSpec.scala]: nitpick: in tests, we usually name objects starting with an uppercase letter... although the object being tested is `commons`, I would probably rename this to match other suites

**Review by kubukoz — COMMENTED** [review]
Looks like the formatter is complaining, can you run `sbt scalafmt` and commit the results? Thanks!

File: modules/decline/test/src/smithy4s/decline/core/commonsSpec.scala (hunk 1 above)
Comment (dwalend) [reply modules/decline/test/src/smithy4s/decline/core/commonsSpec.scala]: And in a PR about camel-case no less.

**Comment by dwalend** [comment]
> > It doesn't look like github linked this PR back to #1539 . What's the right way to make that happen?
> 
> I think you need a hash in the PR title. Ideally, you can say "Closes #1539" and it'll link them so that when the PR gets merged, the issue gets closed.

No luck. I wonder if I had to get it right the first time.

**Review by Baccata — APPROVED** [review]

**Comment by Baccata** [comment]
Thank you !

**Comment by dwalend** [comment]
> Thank you !

Glad to. Next let me see if I can pick up something with a little more smithy4s in it.

**Pull request merged**